### PR TITLE
[KYUUBI #2616] Remove embedded mode support in Kyuubi Hive JDBC driver

### LINKED ISSUE #2616: [Improvement] Remove embedded mode support in Kyuubi Hive JDBC driver
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

In embedded mode, Kyuubi Hive JDBC driver will create a `EmbeddedThriftBinaryCLIService` instead of connecting to Kyuubi server. 

Currently, the Kyuubi Hive JDBC driver uses embedded mode if host and port absent in JDBC url, e.g.

- `jdbc:hive2:///`
- `jdbc:hive2:///dbName;sess_var_list?hive_conf_list#hive_var_list`

### How should we improve?

The embedded mode makes no sense for Kyuubi.
We should remove the support of embedded mode, and change this behavior to use Kyuubi default host(localhost) and default port(10009) if host and port absent.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by jiaoqingbo**
### _Why are the changes needed?_

fix #2616 

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-hive-jdbc/src/test/java/org/apache/kyuubi/jdbc/hive/UtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import java.util.Properties;
+import org.junit.Test;
+
+public class UtilsTest {
+
+  @Test
+  public void testExtractURLComponentsWithAuthoritiesIsEmpty() throws JdbcUriParseException {
+    String uri1 = "jdbc:hive2:///db;k1=v1?k2=v2#k3=v3";
+    String uri2 = "jdbc:hive2:///";
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/test/java/org/apache/kyuubi/jdbc/hive/UtilsTest.java:31]: Add test for `jdbc:kyuubi://` and `jdbc:hive2://`

Comment (jiaoqingbo) [reply kyuubi-hive-jdbc/src/test/java/org/apache/kyuubi/jdbc/hive/UtilsTest.java:31]: ok

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/Utils.java [hunk 2]
```diff
@@ -369,11 +347,13 @@ public static JdbcConnectionParams extractURLComponents(String uri, Properties i
     if ((authorityFromClientJdbcURL == null) || (authorityFromClientJdbcURL.isEmpty())) {
       // Given uri of the form:
       // jdbc:hive2:///dbName;sess_var_list?hive_conf_list#hive_var_list
-      connParams.setEmbeddedMode(true);
-    } else {
-      connParams.setSuppliedURLAuthority(authorityFromClientJdbcURL);
-      uri = uri.replace(authorityFromClientJdbcURL, dummyAuthorityString);
+      authorityFromClientJdbcURL = "localhost:10009";
+      uri =
+          uri.replace(
```
[the remark this answers is not in the document]
Comment (jiaoqingbo) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/Utils.java:352]: ok I will make line351 and line 352 to one line

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**